hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/SystemFileUpdateFacility.java

```diff
@@ -98,49 +105,51 @@ public void handleTxBody(@NonNull final HederaState state, @NonNull final Transa
-                backendThrottle.rebuildFor(throttleManager.throttleDefinitions());
-                frontendThrottle.rebuildFor(throttleManager.throttleDefinitions());
+        if (fileNum == config.feeSchedules()) {
+            return feeManager.update(FileUtilities.getFileContent(state, fileID));
+        } else if (fileNum == config.exchangeRates()) {
+            exchangeRateManager.update(FileUtilities.getFileContent(state, fileID), payer);
+        } else if (fileNum == config.networkProperties()) {
+            final var networkProperties = FileUtilities.getFileContent(state, fileID);
+            final var permissions =
+                    FileUtilities.getFileContent(state, createFileID(config.hapiPermissions(), configuration));
+            configProvider.update(networkProperties, permissions);
+            backendThrottle.applyGasConfig();
+            frontendThrottle.applyGasConfig();
 
-                // Updating the multiplier source to use the new throttle definitions
-                congestionMultipliers.resetExpectations();
-            } else if (fileNum == config.upgradeFileNumber()) {
-                logger.error("Update of file number not implemented");
-            }
-        } catch (HandleException e) {
-            // handle exception suppose to propagate the exception to the caller
-            throw e;
-        } catch (final RuntimeException e) {
-            logger.warn(
-                    "Exception while calling updater for file {}. " + "If the file is incomplete, this is expected.",
-                    fileID,
-                    e);
+            // Updating the multiplier source to use the new gas throttle
+            // values that are coming from the network properties
+            congestionMultipliers.resetExpectations();
+        } else if (fileNum == config.hapiPermissions()) {
+            final var networkProperties =
+                    FileUtilities.getFileContent(state, createFileID(config.networkProperties(), configuration));
+            final var permissions = FileUtilities.getFileContent(state, fileID);
+            configProvider.update(networkProperties, permissions);
+        } else if (fileNum == config.throttleDefinitions()) {
+            final var result = throttleManager.update(FileUtilities.getFileContent(state, fileID));
+            backendThrottle.rebuildFor(throttleManager.throttleDefinitions());
+            frontendThrottle.rebuildFor(throttleManager.throttleDefinitions());
+
+            // Updating the multiplier source to use the new throttle definitions
+            congestionMultipliers.resetExpectations();
+            return result;
         }
+        return SUCCESS;
+    }
+
+    private FileID createFileID(final long fileNum, @NonNull final Configuration configuration) {
+        final var hederaConfig = configuration.getConfigData(HederaConfig.class);
+        return FileID.newBuilder()
+                .realmNum(hederaConfig.realm())
```

Comment:
If these are system files, do we want to constrain realm and shard to `0`?  I don't think any system files are actually allowed to vary between shards and realms.

Reply:
Sharding and realms are still too fuzzy for me to have an opinion. 🙂